logstash-core/lib/logstash/pipeline.rb

```diff
@@ -70,7 +70,7 @@ def non_reloadable_plugins
 
 
   def plugin(plugin_type, name, line, column, *args)
-    @plugin_factory.plugin(plugin_type, name, line, column, *args)
+    @plugin_factory.plugin(plugin_type, name, line, column, "", -1, *args)
```

Comment:
looks really ugly and it feels like these should be optional params e.g. PluginFactory signature would look like:
`plugin(plugin_type, name, line, column, *args, **opts)` (will still get a bit ugly at the Java ext end)

but than from the `def plugin` signature this method was meant to be simply delegating down the line ...

*what I am not sure about, can we really change the underlying signature non backwards-compatibly?*